pkg/cluster/cluster_config.go

```diff
@@ -215,3 +214,15 @@ func getRelease(ctx context.Context, cli client.Client) (Release, error) {
 	initRelease.Version = csv.Spec.Version
 	return initRelease, nil
 }
+
+// IsDefaultAuthMethod returns true if the default authentication method is IntegratedOAuth or empty.
+// This will give indication should Operator should create userGroups or not in the cluster.
+func IsDefaultAuthMethod(ctx context.Context, cli client.Client) (bool, error) {
+	authenticationobj := &configv1.Authentication{}
+	if err := cli.Get(ctx, client.ObjectKey{Namespace: "", Name: "cluster"}, authenticationobj); err != nil {
+		// here: true or false not matter once error is returned
+		return true, fmt.Errorf("failed to get Authentication CR cluster: %w", err)
+	}
+
+	return (authenticationobj.Spec.Type == "IntegratedOAuth" || authenticationobj.Spec.Type == ""), nil
```

Comment:
it appears there is a constant for this `configv1.AuthenticationTypeIntegratedOAuth`

Reply:
i was thinking to make it:
`return (authenticationobj.Spec.Type != "OIDC"), nil`

or should we only create if it is `AuthenticationTypeIntegratedOAuth` ?

Reply:
I don't have any strong opinion but sinceI don't know what are all the possible values (i.e. behind a feature flag like the `OIDC` one) I would check if the type is among the supported values `IntegratedOAuth` or ""  to be on the safe side